docs/dotnet/dotnet-http-metrics.md

```diff
@@ -221,18 +261,175 @@ All Http server metrics are reported by `Microsoft.AspNetCore.Hosting ` meter.
 
 ### Metric: `http.server.duration`
 
-**TODO: should we add error status to http semconv ?**  - https://github.com/open-telemetry/opentelemetry-specification/issues/3243
 **TODO: Opt-in address and port?**
 **TODO: method cardinality?**
 
 Follows common [http.server.duration](../http/http-metrics.md#metric-httpserverduration)
 
+<!-- semconv metric.dotnet.http.server.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.server.duration` | Histogram | `s` | Measures the duration of inbound HTTP requests. |
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.server.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `dotnet.error.code` | string | General-purpose error code reported by .NET, as a starter it supports terminal statuses of .NET task https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskstatus. | `Canceled`; `RanToCompletion` | Recommended |
+| `dotnet.http.request.error` | string | HTTP Request error as defined inhttps://github.com/dotnet/runtime/blob/c430570a01c103bc7f117be573f37d8ce8a129b8/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs | `extended_connection_not_supported`; `http_protocol` | Recommended |
```

Comment:
> I am not sure what role it is expected to play.

Agreed. I think trying to introduce a "general purpose error code" is wrong. It's feels like if we mixed all kinds of unrelated error enums into a single enum and also added TaskStatus there which is an orthogonal concept ie. a `TaskStatus.Faulted` indicates some error reason. Also note that `TaskStatus.Cancelled` can mean both user cancellation and timeout, and SocketsHttpHandler [has a mechanism to distinguish them](https://learn.microsoft.com/en-gb/dotnet/fundamentals/networking/http/httpclient#http-error-handling), so it might be valuable to report different values for timeout and cancellation.